Assets/Scripts/Augment/AugmentImplementations/Solar/SolarBody.cs

```diff
@@ -8,14 +8,22 @@ public class SolarBody : GunBody
     [SerializeField]
     private MeshRenderer meshRenderer;
 
+    [SerializeField]
+    private Transform rayCastOrigin;
+
+    [SerializeField]
+    private LayerMask obscuringLayers;
+
     private Material solarPanelMaterial;
 
     [SerializeField, Range(0, 1)]
     protected float reloadEfficiencyPercentagen = 0.1f;
 
-    private const float sunDirection = 90f;
-    private const float sunDirectionSpan = 45f;
+    private Quaternion globalLightDirection;
+
+    private const float maxObscuringCheckDistance = 15f;
     private const float coolDownSeconds = 0.5f;
+    private const float chargeUpSeconds = 1; 
```

Comment:
When the lerp goes on for a whole second I can see why it would not make the point of change clear to the user, but you could lerp for like 100ms or less and it should still be noticeable :)

Though I'm not going to prevent you from pressing the big green button this time.

Reply:
I tested with separate values for both, but yeah imma merge lol